src/crepe/api/SceneManager.hpp

```diff
@@ -4,12 +4,15 @@
 
 namespace crepe {
 
-template <typename T>
-void SceneManager::add_scene() {
+template <typename T, typename... Args>
+void SceneManager::add_scene(Args &&... args) {
 	using namespace std;
 	static_assert(is_base_of<Scene, T>::value, "T must be derived from Scene");
 
-	Scene * scene = new T(this->component_manager);
+	Scene * scene = new T(std::forward<Args>(args)...);
```

Comment:
Kan dit zonder een `new` gedaan worden. want stel `scene->component_manager = this->component_manager` gaat fout heb je een memory leak omdat je eerst een `new` doet

Reply:
Daar heb je absoluut een punt. Liever maak ik direct een `unique_ptr` aan, maar omdat alleen `SceneManager` een `Scene` aan mag maken, moet ik eerst `new` gebruiken en kan ik het geheel daarna pas in een `unique_ptr` gieten. Het is wellicht wel verstandig om `scene->component_manager = this->component_manager ` pas aan te roepen nadat het een `unique_ptr` is geworden (dit zal ik later aanpassen, ik zit nu namelijk op m’n telefoon te reageren).